test/subscription_map.hpp

```diff
@@ -472,16 +480,24 @@ class multiple_subscription_map
 
         auto& subscription_set = h_iter->second.value;
 
-        auto new_pair = std::make_pair(key, std::forward<V>(value));
-        auto insert_result = subscription_set.insert(new_pair);
-        if (insert_result.second) {
+        // I have to do a find first, because insert_or_assign is available in C++17 only
+        // https://www.fluentcpp.com/2018/12/11/overview-of-std-map-insertion-emplacement-methods-in-cpp17/
+
+        auto existing_iter = subscription_set.find(key);
+        if(existing_iter == subscription_set.end()) {
+            subscription_set.insert(std::make_pair(std::forward<Key>(key), std::forward<V>(value)));
```

Comment:
It doesn't support an emplace function? The call to std::make_pair is required?

Reply:
It is replaced now by emplace with piecewise construct